lib/cocoapods/installer/pod_source_installer.rb

```diff
@@ -110,18 +112,24 @@ def download_source
         end
       end
 
-      # Verify the source of the spec is secure, which is used to
-      # show a warning to the user if that isn't the case
-      # This method doesn't verify all protocols, but currently
-      # only prohibits unencrypted http:// connections
+      # Verify the source of the spec is secure, which is used to show a warning to the user if that isn't the case
+      # This method doesn't verify all protocols, but currently only prohibits unencrypted 'http://' and 'git://''
+      # connections.
+      #
+      # @return [void]
       #
       def verify_source_is_secure(root_spec)
-        return if root_spec.source.nil? || root_spec.source[:http].nil?
-        http_source = URI(root_spec.source[:http])
-        return if http_source.scheme == 'https' || http_source.scheme == 'file'
-        UI.warn "'#{root_spec.name}' uses the unencrypted http protocol to transfer the Pod. " \
+        return if root_spec.source.nil? || (root_spec.source[:http].nil? && root_spec.source[:git].nil?)
+        source = if !root_spec.source[:http].nil?
+                   URI(root_spec.source[:http])
+                 elsif !root_spec.source[:git].nil?
+                   URI(root_spec.source[:git].to_s)
```

Comment:
Why does `:git` need `to_s` but `:http` does not?

Reply:
@amorde good Q, for some of the tests and podspecs the `git` value resolves to a `Pathname`... I could also make the `:http` one to use `to_s` it should be harmless.